src/table.jl

```diff
@@ -54,21 +72,22 @@ end
 
 Tables.partitions(x::Stream) = x
 
-Stream(io::IO, pos::Integer=1, len=nothing; convert::Bool=true) = Stream(Base.read(io), pos, len; convert=convert)
-Stream(str::String, pos::Integer=1, len=nothing; convert::Bool=true) = isfile(str) ? Stream(Mmap.mmap(str), pos, len; convert=convert) :
-    throw(ArgumentError("$str is not a file"))
+Stream(input, pos::Integer=1, len=nothing; kw...) = Stream([ArrowBlob(tobytes(input), pos, len)]; kw...)
+Stream(input::Vector{UInt8}, pos::Integer=1, len=nothing; kw...) = Stream([ArrowBlob(tobytes(input), pos, len)]; kw...)
+Stream(inputs::Vector; kw...) = Stream([ArrowBlob(tobytes(x), 1, nothing) for x in inputs]; kw...)
 
 # will detect whether we're reading a Stream from a file or stream
-function Stream(bytes::Vector{UInt8}, off::Integer=1, tlen::Union{Integer, Nothing}=nothing; convert::Bool=true)
-    len = something(tlen, length(bytes))
-    if len > 24 &&
-        _startswith(bytes, off, FILE_FORMAT_MAGIC_BYTES) &&
-        _endswith(bytes, off + len - 1, FILE_FORMAT_MAGIC_BYTES)
-        off += 8 # skip past magic bytes + padding
-    end
+function Stream(inputs::Vector{ArrowBlob}; convert::Bool=true)
     dictencodings = Dict{Int64, DictEncoding}() # dictionary id => DictEncoding
     dictencoded = Dict{Int64, Meta.Field}() # dictionary id => field
-    batchiterator = BatchIterator(bytes, off)
+    blob = inputs[1]
```

Comment:
why pass a vector if it's just going to be accessed by index?

Reply:
I mean, there isnt' really an `Indexable` interface to "loosen" the type restriction here, so a `Vector` is a natural candidate. We could probably loosen to `AbstractArray`, but that's an easy change if someone needs to pass something other than an array.